packages/flutter/lib/src/widgets/single_child_scroll_view.dart

```diff
@@ -700,19 +685,34 @@ class _RenderSingleChildViewport extends RenderBox with RenderObjectWithChildMix
   @override
   Rect describeSemanticsClip(RenderObject child) {
     assert(axis != null);
-    switch (axis) {
-      case Axis.vertical:
+    final double remainingOffset = _maxScrollExtent - offset.pixels;
+    switch (axisDirection) {
+      case AxisDirection.up:
         return Rect.fromLTRB(
           semanticBounds.left,
-          semanticBounds.top - cacheExtent,
+          semanticBounds.top - remainingOffset,
           semanticBounds.right,
-          semanticBounds.bottom + cacheExtent,
+          semanticBounds.bottom + offset.pixels,
         );
-      case Axis.horizontal:
+      case AxisDirection.right:
+        return Rect.fromLTRB(
+          semanticBounds.left - offset.pixels,
+          semanticBounds.top,
+          semanticBounds.right + remainingOffset,
+          semanticBounds.bottom,
+        );
+      case AxisDirection.down:
+        return Rect.fromLTRB(
+          semanticBounds.left,
+          semanticBounds.top - offset.pixels,
+          semanticBounds.right,
+          semanticBounds.bottom + remainingOffset,
```

Comment:
Isn't this making the semantics clip larger than necessary? AFAK, "remainingOffset" is the distance from the top of the viewport to the end of the scrollable content, so the semantics clip here would be one viewport height to tall, no?

Not that it matters in practice, but may cause confusion in the future...

Reply:
Could we actually add a test for this? The height of the semanticsClip rect should be exactly the height of the scrollable content, right?

Reply:
Ah, wait: `_maxScrollExtent` is already accounting for the viewport width. Still, I think adding a test like I described above would be great.